cmd/cluster/azure/create.go

```diff
@@ -100,6 +110,22 @@ func (o *RawCreateOptions) Validate(ctx context.Context, opts *core.CreateOption
 	if o.DiskEncryptionSetID != "" && o.ResourceGroupName == "" {
 		return nil, fmt.Errorf("--resource-group-name is required when using --disk-encryption-set-id")
 	}
+
+	// Validate publisher, offer, sku, and version flags are provided if one of them is not empty
+	marketplaceImageInfo := map[string]string{
+		"marketplace-publisher": o.MarketplacePublisher,
+		"marketplace-offer":     o.MarketplaceOffer,
```

Comment:
please include test cases for this, see TestCreateCluster

Reply:
unit test added